adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralInterstitialRenderer.m

```diff
@@ -0,0 +1,151 @@
+                                                 completionHandler{
+    _adConfiguration = adConfiguration;
+    // Store the ad config and completion handler for later use.
+    __block atomic_flag completionHandlerCalled = ATOMIC_FLAG_INIT;
+    __block GADMediationInterstitialLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationInterstitialAdEventDelegate>(
+                                                                        _Nullable id<GADMediationInterstitialAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationInterstitialAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *unitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+    
+    // if the unitId is nil.
+    if ([GADMAdapterMintegralUtils isEmpty:unitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Unit ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if ([GADMAdapterMintegralUtils isEmpty:adConfiguration.bidResponse]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"bid token cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    _interstitialAd = [[MTGNewInterstitialBidAdManager alloc]initWithPlacementId:placementId unitId:unitId delegate:self];
+    [_interstitialAd loadAdWithBidToken:adConfiguration.bidResponse];
+}
+- (void)newInterstitialBidAdLoadSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+}
+
+
+- (void)newInterstitialBidAdResourceLoadSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+
+- (void)newInterstitialBidAdLoadFail:(nonnull NSError *)error adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+- (void)newInterstitialBidAdShowSuccess:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+    [_adEventDelegate reportImpression];
+    [_adEventDelegate willPresentFullScreenView];
+}
+
+- (void)newInterstitialBidAdShowSuccessWithBidToken:(nonnull NSString * )bidToken adManager:(MTGNewInterstitialBidAdManager *_Nonnull)adManager {
+}
```

Comment:
What are the differences between these two callbacks? How did you decide in which callback do we call the below GMA SDK callbacks?
```
    [_adEventDelegate reportImpression];
    [_adEventDelegate willPresentFullScreenView];
```

Reply:
mintegral present the full screen ad and reports the impression when it receives the newInterstitialBidAdShowSuccess callback method.